src/watcher.js

```diff
@@ -301,6 +313,29 @@ class Watcher extends SafeEventEmitter {
 
         this.on('load.vod', () => observe(vodChatWatcher, $('.vod-chat')[0]));
     }
+
+    directoryObserver() {
+        const observe = (watcher, selector) => {
+            // timeout used because events sometimes fire before elements are created
+            setTimeout(() => {
+                const element = $(selector)[0];
+                if (!element) return;
+                if (watcher) watcher.disconnect();
+                watcher.observe(element, {childList: true, subtree: true});
+            }, 200);
+        };
+
+        directoryWatcher = new window.MutationObserver(() => {
+            this.emit('directory.changed');
```

Comment:
you should be watching for specific elements being created (i.e. creation of the vodcast streams), not firing on every change